### DISCUSSION TITLE
Issues with the .not() filter in javascript

### DISCUSSION THREAD

**Discussion by LankieB** [supabase/supabase#12528] [Questions]
Please has anyone figured out how to use the negative filter properly?
![Screenshot 2023-02-19 at 07 51 46](https://user-images.githubusercontent.com/64866720/219933914-575d7ab0-7e5d-4a2f-a55c-e0e88d987757.png)

I want to limit select results to items that excludes certain user ids. And I have the user id's in UUID form.

full code is below:

.from("profile")
    .select()
    .not("id", "in", ('"0000f317-7003-4db2-b117-72189260686f", "000bfd34-a1ae-44f2-94de-94d8b73a3eeb", "00342f6c-8c3f-4480-90c5-35ff61e6edda")
 
However, it never works. I have tried removing the quotes, tried passing the excluded items as an array etc. But it is still not working.

Please is there a way to do this?

**Comment by GaryAustin1** [comment]
```
.not("id","in",'( "UUID1","UUID2" )' )
```
Should work.

Note that is you have nulls for id they will not show up in a not filter.  Nulls "don't exist" in Postgres so are ignored completely except using the `is` filter.

**Reply by LankieB** [reply]
Thanks so much Gary for the prompt reply.

Tried it, but apparently still not working. I am probably still missing something. It only works for the first UUID in the (brackets). 

Whilst I waited, I used another method, chaining neq's with the IDs and this worked.

Sharing below in full in case it helps someone else.

// create initial query 
const supabaseQuery = supabase.select()........ [and other conditions] 

const idArrayToIgnore = [array of UUIDs];

let filteredQuery = supabaseQuery;

// limit the query to only items that are not in the connections list
  for (let i = 0; i < idArrayToIgnore.length; i++) {
     filteredQuery = filteredQuery.neq("id", idArrayToIgnore.length[i]);
  }
  
// run query and get results from supabase
const (data, error) = await supabase (....)

**Comment by yan-hic** [comment]
I agree with the OP that passing an array of values is error-prone. Using `()` is counter-intuitive as arrays use square brackets. Ideally the param should accept an JSON array if operator is `in` i.e. one returned by `JSON.stringify`. 

Getting to the supported format almost always requires some sort of mapping, replace and/or stringify. Could be made simpler by wrapping behind the scenes.
